src/Compiler/Driver/fsc.fs

```diff
@@ -842,6 +842,22 @@ let main3
             errorRecoveryNoRange e
             exiter.Exit 1
 
+    let refAssemblySignatureHash =
+        match tcConfig.emitMetadataAssembly with
+        | MetadataAssemblyGeneration.None -> None
+        | MetadataAssemblyGeneration.ReferenceOnly
+        | MetadataAssemblyGeneration.ReferenceOut _ ->
+            try
+                Fsharp.Compiler.SignatureHash.calculateSignatureHashOfFiles
+                    typedImplFiles
+                    tcGlobals
+                    Fsharp.Compiler.SignatureHash.PublicOnly
+                + Fsharp.Compiler.SignatureHash.calculateHashOfAssemblyTopAttributes topAttrs tcConfig.platform
+                |> Some
+            with e ->
+                printfn "Unexpected error when hashing implied signature, will hash the all of .NET metadata instead. Error: %O " e
```

Comment:
It's an internal detail, I did not want to have this leaked to the customer - worst case, everything gets hashed as before.
Warning might signalize something is wrong in their code, but it will not be the case...

Reply:
Would be nice if we could send telemetry when this happens. But that's probably out of scope here.

Reply:
Well this is the fsc.exe, I am not sure we even have telemtry hooked up - we only have it for the VS Editor actions at the moment.

I am very open to any kind of suggestion what to do.
This is the behaviour I want to have:

- It should not a fail a build, falling back to old logic must be preferred over crashing
- If it happens, it is definitely our/my fault -> it must be clear to the person on screen that the issue is not with the code they have written

Reply:
Yeah that's why I thought telemetry would be great. But don't know what it would take to send it from fcs. 

Maybe when we use this as part of incremental compilation we will be able to send failures through VS telemetry.